common/src/main/java/io/homeassistant/companion/android/common/data/prefs/WearPrefsRepository.kt

```diff
@@ -7,10 +7,10 @@ interface WearPrefsRepository {
     suspend fun removeTileShortcuts(tileId: Int?): List<String>?
     suspend fun getShowShortcutText(): Boolean
     suspend fun setShowShortcutTextEnabled(enabled: Boolean)
-    suspend fun getTemplateTileContent(): String
-    suspend fun setTemplateTileContent(content: String)
-    suspend fun getTemplateTileRefreshInterval(): Int
-    suspend fun setTemplateTileRefreshInterval(interval: Int)
+    suspend fun getAllTemplateTiles(): Map<Int?, Pair<String, Int>>
+    suspend fun getTemplateTile(tileId: Int?): Pair<String, Int>?
+    suspend fun setTemplateTile(tileId: Int?, content: String, refreshInterval: Int)
+    suspend fun removeTemplateTile(tileId: Int?): Pair<String, Int>?
```

Comment:
Initially I wanted to expose the `Pair<String, Int>` as a proper data class, but I'm not sure where to put it. It needs to be in `common` because this interface is in `common`, but I couldn't find a good place to put it in. Hence I went with an internal data class in `WearPrefsRepositoryImpl` and exposing the data as `Pair`.

Reply:
You can clarify it by adding documentation to functions, for example:

```kotlin
/** @return [Pair] containing the template content and the refresh interval, or `null` if it doesn't exist */
suspend fun getTemplateTile(tileId: Int?): Pair<String, Int>?
```

Or you could create an `impl.entities` package like authentication/integration/websocket are doing and add it there.

Reply:
> Or you could create an `impl.entities` package like authentication/integration/websocket are doing and add it there.

Ah, thanks, that could work :+1: Being able to use the real field names instead of `first` and `second` makes the code so much easier to read IMO, so I'll go with that.